src/read-only-handler.ts

```diff
@@ -85,16 +83,20 @@ export class ReadOnlyHandler {
             return desc;
         }
         const shadowDescriptor = getOwnPropertyDescriptor(shadowTarget, key);
-        if (!desc.configurable && !shadowDescriptor) {
+        if (!isUndefined(shadowDescriptor)) {
+            return shadowDescriptor;
+        }
+        desc = wrapDescriptor(membrane, desc, getReadOnlyValue);
+        delete desc.set; // readOnly membrane does not allow setters
```

Comment:
Performance wise, deleting the `set` key in the descriptor change the hidden class of the object, which is not perfect. What about adding a setter that throws a generic error?

Reply:
not a big deal since descriptor objects are created every time by the engine.